authz/sdk_server_interceptors.go

```diff
@@ -73,3 +82,86 @@ func (i *StaticInterceptor) StreamInterceptor(srv interface{}, ss grpc.ServerStr
 	}
 	return handler(srv, ss)
 }
+
+// FileWatcherInterceptor contains details used to make authorization decisions
+// by watching a file path that contains authorization policy in JSON format.
+type FileWatcherInterceptor struct {
+	internalInterceptor unsafe.Pointer // *StaticInterceptor
+	policyFile          string
+	policyContents      []byte
+	refreshDuration     time.Duration
+	cancel              context.CancelFunc
+}
+
+// NewFileWatcher returns a new FileWatcherInterceptor from a policy file
+// that contains JSON string of authorization policy and a refresh duration to
+// specify the amount of time between policy refreshes.
+func NewFileWatcher(file string, duration time.Duration) (*FileWatcherInterceptor, error) {
+	if file == "" {
+		return nil, fmt.Errorf("authorization policy file path is empty")
+	}
+	if duration <= time.Duration(0) {
+		return nil, fmt.Errorf("requires refresh interval(%v) greater than 0s", duration)
+	}
+	i := &FileWatcherInterceptor{policyFile: file, refreshDuration: duration}
+	i.updateInternalInterceptor()
```

Comment:
Is the constructor supposed to error if the file is invalid?  I thought that's what we decided, but now I don't remember for sure.

Reply:
Oh I missed that in my last commit, when I refactored for avoiding repetitive code. And updated the tests too. Thank you!